### DISCUSSION TITLE
How to create a new column copying an existing one?

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#3495] [Q&A] [answered]
Hi,
I'm trying to add a new field to my ls command using

```
ls | each { insert newField $it.name }
```

but I have empty result.

**Comment by sophiajt** [comment, marked as answer]
@aborruso - I think the current `insert` takes a block that it runs to get the value to insert. So instead of what you have, try this:

```
ls | each { insert newField { $it.name } }
```
in the upcoming 0.32

or 

```
ls | each { insert newField { = $it.name } }
```
in 0.31 and earlier

**Reply by aborruso** [reply]
> ls | each { insert newField { $it.name } }

If I run it I have

```
error: Dynamic commands must start with a block (or variable pointing to a block)
  ┌─ shell:1:31
  │
1 │ ls | each { insert newField { $it.name } }
  │                               ^^^^^^^^ needs to be a block
```

**Reply by sophiajt** [reply]
If you're running a version earlier than the upcoming 0.32, you'll need to use the second example above

**Reply by aborruso** [reply]
Thank you very much
